lib/shipit/command.rb

```diff
@@ -107,16 +107,15 @@ def stream(&block)
       begin
         read_stream(@out, &block)
       rescue TimedOut => error
-        @code = 'timeout'
         yield red("No output received in the last #{timeout} seconds.") + "\n"
         terminate!(&block)
         raise error
       rescue Errno::EIO # Somewhat expected on Linux: http://stackoverflow.com/a/10306782
       end
 
-      _, status = Process.waitpid2(@pid)
-      @code = status.exitstatus
       self
+    ensure
+      reap_child!
```

Comment:
`stream` is where `start` is called, hence the blocking reap call.

Reply:
Will this work if `start` raises?

Reply:
Yes, this is covered with 2 existing tests that cause permission denied and file not found exceptions.